debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresTaskContext.java

```diff
@@ -100,11 +100,17 @@ private SlotState getCurrentSlotState(PostgresConnection connection) throws SQLE
     }
 
     protected ReplicationConnection createReplicationConnection(boolean exportSnapshot) throws SQLException {
+        final boolean dropSlotOnStop = config.dropSlotOnStop();
+        if (dropSlotOnStop) {
+            LOGGER.warn(
+                    "Connector has enabled automated replication slot removal upon restart ({} = true). This setting can lead to data loss in production environments!",
```

Comment:
"Data loss" isn't the right word, as your data still will be in the DB. But it may mean that after a downtime to connector cannot resume from the previous offset. I'll reword a bit and merge. Thanks!